types/src/system/auction/mod.rs

```diff
@@ -98,9 +98,9 @@ pub trait Auction:
         amount: U512,
     ) -> Result<U512, Error> {
         let account_hash = AccountHash::from_public_key(&public_key, |x| self.blake2b(x));
-        if self.get_caller() != account_hash {
-            return Err(Error::InvalidPublicKey);
-        }
+        if self.get_immediate_caller() != Some(&CallStackElement::Session { account_hash }) {
```

Comment:
If in my deploy, I call a contract that calls an auction endpoint on my behalf – will that call fail now?

Reply:
> If, in my deploy's session code, I call a contract that calls an auction endpoint on my behalf – will that call fail now? 

Yes.

> Alternatively, is it only possible to call auction contract directly now? 

Yes.

> Will things like `auction_manager` (that cspr.live is using for delegate/undelegate) contract stop working?

Yes.

Further discussion on the subject at standup.